src/ert/gui/ertwidgets/models/storage_model.py

```diff
@@ -165,45 +169,59 @@ def _load_storage(self, storage: Storage) -> None:
 
             self._children.append(ex)
 
-    @staticmethod
-    def columnCount(parent: QModelIndex) -> int:
+    @override
+    def columnCount(self, parent: QModelIndex = default_index) -> int:
         return _NUM_COLUMNS
 
-    def rowCount(self, parent: QModelIndex) -> int:
+    @override
+    def rowCount(self, parent: QModelIndex = default_index) -> int:
         if parent.isValid():
             if isinstance(parent.internalPointer(), RealizationModel):
                 return 0
             return len(parent.internalPointer()._children)
         else:
             return len(self._children)
 
-    def parent(self, index: QModelIndex) -> QModelIndex:
-        if not index.isValid():
+    @overload
+    def parent(self, child: QModelIndex) -> QModelIndex: ...
+    @overload
+    def parent(self) -> Optional[QObject]: ...
+    @override
+    def parent(self, child: QModelIndex = default_index) -> Optional[QObject]:
+        if not child.isValid():
```

Comment:
Why does this function return Optional[QObject] ?
As far as I can see this returns a QModelIndex on all exits?

Reply:
That is true, but the type checker does not like that. I believe this has to do with overloads as a similar situation works correctly when there is no overload involved.

Overloads are somewhat of a special case. Here you would have to figure out that

```python
    def parent(self, child: Optional[QModelIndex] = None) -> QModelIndex:
```

satisfies the following interface:

```python
    @overload
    def parent(self, child: QModelIndex) -> QModelIndex: ...
    @overload
    def parent(self) -> Optional[QObject]: ...
```

which is quite complicated when you take into account all possibilities. mypy will not check that you always get a QModelIndex when the parameter `child` is supplied for instance, but it does check that the signature for the implementation accepts all the parameters in the overload defintions.